README.md

```diff
@@ -500,6 +500,75 @@ All non-alpha releases derive from tagged commits on `master`. The tag defines t
+
+### `Equinox.Cosmos.AccessStrategy`
+
+TL;DR `Equinox.Cosmos`: (see also: [the storage model](DOCUMENTATION.md#Cosmos-Storage-Model) for a deep dive, and [glossary, below the table](#access-strategy-glossary) for definition of terms)
+- keeps all the events for a stream in a single single [CosmosDB _logical partition_](https://docs.microsoft.com/en-gb/azure/cosmos-db/partition-data)
+- always has a special 'index' document (we term it the `Tip` document), per logical partition/stream that is accessible via an efficient _point read_ (it always has a CosmosDB `id` value of `"-1"`)
+- Events are stored in immutable documents in the logical partition, the `Tip` document is the only document that's ever updated. 
+- all writes touch the `Tip`, so we have a natural way to invalidate any cached State for a given Stream - we retain the `_etag` of the `Tip` document, and updates (or consistent reads) are contingent on it not having changed
+- Concurrency control of updates is by virtue of the fact that every update to the `Tip` touches the document _and thus alters (invalidates) the `_etag` value_. This means we don't rely on uniqueness constraints blocking inserts of Event documents and/or having to add a new Event every time we are updating something. (This fact is crucial for the `RollingState` and `Custom` strategies).
+- The `Tip` document, (and the fact we hold its `_etag` in our cache alongside the State we have derived from the Events), is at the heart of why consistent reads are guaranteed to be efficient (Equinox does the read of the `Tip` document contingent on the `_etag` not having changed; a read of any size costs only 1 RU if the result is `304 NOT Modified`)
+- Specific Access Strategies:
+  - define what we put in the `Tip`
+  - control how we short circuit loading if we have a Snapshot or Reset Event
+  - allow one to post-process the events we are writing as required for reasons of optimization
+
+#### `Cosmos` Access Strategy overviews
+
+| Strategy | TL;DR | `Tip` document maintains | Best suited for |
+| :--- | :--- | :--- | :--- |
+| `Unoptimized` | Keep Events only (but there's still an (empty) `Tip` document) | Count of event (and ability to have any insertion invalidate the cache for any reader) | No load, event counts or event sizes worth talking about. <br/> initial implementations. |
+| `LatestKnownEvent` | Special mode for when every event is completely independent | A copy of the most recent event, together with the count | Tracking changes to a document where we're not really modelling events as such, but with optimal efficiency (every read is a point read of a single document) |
+| `Snapshot` | Keep a (single) snapshot in `Tip` at all times, guaranteed to include _all_ events | A single unfold produced by `toSnapshot` based on the `state'` (i.e., including any events being written) every time, together with the event count | Typical event-sourced usage patterns. <br/> Good default approach. |
```

Comment:
When Snapshot strategy, is every writes updates the `Tip` file with the current snapshot and the event version? Based on what I have seen, I see the `Tip` get updated with the event number and `c": "PickTicketSnapshot"`, but I don't see the snapshot/state been persisted. Not sure if that is expected. If yes, my question is how does this help with a long-running stream, since it feels like it has to build the state from scratch anyway. If not, I think I am problably doing something wrong, so would potentially need some help on this.

Reply:
Actually, is `d` storing the compressed state?

Reply:
Snapshot means events and snapshots (RollingState is only updating Tip)
the snapshot itself is base64d + compressed when its in the Tip (will add a mention of that above) you can decode it with https://jgraph.github.io/drawio-tools/tools/convert.html (click decode) - maybe its not

Reply:
added a sentence in preamble

Reply:
yes u.[d is compressed snapshot data